src/native/libs/System.Native/pal_autoreleasepool.m

```diff
@@ -3,27 +3,35 @@
 
 #include "pal_autoreleasepool.h"
 #include <Foundation/Foundation.h>
+#include <objc/runtime.h>
 
-@interface PlaceholderObject : NSObject
-- (void)noop:(id)_;
-@end
-
-@implementation PlaceholderObject : NSObject
-- (void)noop:(id)_
+static void noop_release(id self, SEL _cmd)
 {
     [self release];
 }
-@end
 
 void EnsureNSThreadIsMultiThreaded(void)
 {
     if (![NSThread isMultiThreaded])
     {
+        NSString *placeholderClassName = @"PlaceholderObject";
+        NSString *uuidClassName = [[NSUUID UUID] UUIDString];
+        const char* className = [[placeholderClassName stringByAppendingString:uuidClassName] UTF8String];
+
+        // For the overwhelming majority of all cases, there will only ever be one runtime at a time in a process. However,
+        // for library mode both in NativeAOT and Mono, more than one library can be used. Therefore, 
+        // the placeholder class needs to be registered at runtime and its name unique in the event (unlikely as it may be)
+        // this is called more than once. 
+        Class defPlaceholderObject = objc_allocateClassPair([NSObject class], className, 0);
+        class_addMethod(defPlaceholderObject, @selector(noop:), (IMP)noop_release, "v@:");
+        objc_registerClassPair(defPlaceholderObject);
+
+        id placeholderObject = [[defPlaceholderObject alloc] init];
+
```

Comment:
Might be able to avoid the placeholder class by using an existing class. The only thing it needs is a selector that takes a single parameter and doesn't return a value.

For instance: `NSMutableString appendString:`

So this would be:

```objective-c
id placeholderObject = [[NSMutableString alloc] init];		
[NSThread detachNewThreadSelector:@selector(appendString:) toTarget:placeholderObject withObject:@""];
[placeholderObject release];
```

Reply:
We do the same thing in xamarin-macios, so this would only apply when not using our SDKs. I have no idea which platforms that would be though (should be easy enough to test in a simple Xcode app by just printing `[NSThread isMultiThreaded]` as the first line of `main`.